gitCoreJGit/src/main/java/com/virtuslab/gitcore/impl/jgit/GitCoreRepository.java

```diff
@@ -214,18 +249,16 @@ public ObjectId gitCoreCommitToObjectId(IGitCoreCommit commit) throws GitCoreExc
       @Unique RevCommit mergeBase = walk.next();
       LOG.debug(() -> "Detected merge base for ${c1.getHash().getHashString()} " +
           "and ${c2.getHash().getHashString()} is ${mergeBase.toString()}");
-      return mergeBase;
-    } catch (IOException e) {
-      throw new GitCoreException(e);
-    }
+      GitCoreCommitHash mergeBaseHash = mergeBase != null ? new GitCoreCommitHash(mergeBase.getId().getName()) : null;
```

Comment:
maybe inline var?

Reply:
Ok, I'll change on one of downstream PRs